scripts/port-forward-all.sh

```diff
@@ -1,10 +1,15 @@
 #!/bin/bash
 
+read -p "Enter BOOKBUYER pod local port:" -i "8080" -e BOOKBUYER_PORT
+read -p "Enter BOOKSTOREv1 pod local port:" -i "8081" -e BOOKSTOREv1_PORT
+read -p "Enter BOOKSTOREv2 pod local port:" -i "8082" -e BOOKSTOREv2_PORT
+read -p "Enter BOOKTHIEF pod local port:" -i "8083" -e BOOKTHIEF_PORT
+
 ./scripts/port-forward-zipkin.sh &
-./scripts/port-forward-bookbuyer-ui.sh &
-./scripts/port-forward-bookstore-ui-v2.sh &
-./scripts/port-forward-bookstore-ui-v1.sh &
-./scripts/port-forward-bookthief-ui.sh &
+./scripts/port-forward-bookbuyer-ui.sh $BOOKBUYER_PORT &
+./scripts/port-forward-bookstore-ui-v2.sh bookstore-v2 $BOOKSTOREv2_PORT &
+./scripts/port-forward-bookstore-ui-v1.sh bookstore-v1 $BOOKSTOREv1_PORT &
+./scripts/port-forward-bookthief-ui.sh $BOOKTHIEF_PORT &
```

Comment:
Since these are being backgrounded and run somewhat in parallel, how would the prompts look like? Can we make this simpler:
- ./scripts port-forward-bookbuyer.sh will run the port forward on the default ports
- ./scripts-port-forward-bookbuyer.sh 7071 will override the port used for port forwarding

This will avoid the need for prompts and seems like a simpler solution. What do you think?

Reply:
Appreciate the quick feedback!  Sure thing.  I haven't experienced an issue with the prompts but it's probably better to get rid of them.  Do you have any suggestions for the port-forward-all.sh script?  Should I just leave it as is and if someone needs to customize the ports, then they can edit the script manually?

Reply:
Another option is to keep the code here as is, and make use of an environment variable in the service specific port forward scripts, something like
```
BOOKBUYER_LOCAL_PORT="${BOOKBUYER_LOCAL_PORT:-8080}"
POD="$(kubectl get pods --selector app=bookbuyer -n "$BOOKBUYER_NAMESPACE" --no-headers | grep 'Running' | awk '{print $1}')"

kubectl describe pod "$POD" -n "$BOOKBUYER_NAMESPACE"

kubectl port-forward "$POD" -n "$BOOKBUYER_NAMESPACE" $BOOKBUYER_LOCAL_PORT:80
```

Someone wanting to override this port could use `BOOKBUYER_LOCAL_PORT=7070 ./scripts/port-forward-all.sh`.
This would apply to the other `port-forward-*ui*.sh` scripts